### DISCUSSION TITLE
Approved tutorial somewhere?

### DISCUSSION THREAD

**Discussion by BassXX** [awawa-dev/HyperHDR#321] [Q&A]
Hi there,

I followed some random tutorial on the interent. Now, after reading HyperHDR basics: do and DON'T (https://github.com/awawa-dev/HyperHDR/discussions/320) I realized, that i did quite a few things wrong.

- I have shared PSU for Pi and LED Strip
- I try to drive my SK6812 from my pi directly

Are there any "approved" or "official" tuorials out there somewhere, so I can avoid these inconveniences?

Thanks in advance.

**Comment by awawa-dev** [comment]
Hi
So probably 3,3 to 5V level shifter is also missing? If it's working without random flashing then fine but it's out of the technical specification and it's matter of luck... But probably the main problem is your shared power supply that may cause under-voltage events on the Rpi and malfunction of the USB grabber (permanent or just from the time to time) or other peripheral devices. 

So far I didn't find a tutorial that covers HyperHDR sk6812 in full aspect, I have got my own tutorial on the blog but it's a bit outdated (alternative sacrifice LED level shifter was used and sk6812 RGBW neutral white, nevertheless it's still made as it should be). Probably I'll update it after v18 is released, if nothing new comes from the community.

**Reply by BassXX** [reply]
Yes it works, but very random only (every 3-5 reboots).

Will check that i can plugthe pi with its own dedicated PSU.

Also the Level Shifter is missing. Never heard of it before since it was not mentioned in the turorial i followed.

Is there any specific info on how to build/wire it? 
Or are there ready-to-go components?

Thanks & Regards

**Reply by awawa-dev** [reply]
What does it mean? Your all system works including USB grabber and then completely stops? Or just the USB grabber stops working but for example animations run from HyperHDR still work with your LED strip? If you are using dedicated PSU remember also about the common ground. The wiring for recommended level shifter is here https://github.com/awawa-dev/HyperHDR/wiki/Level-Shifter all you need is to provide +5V, two(!) grounds, data input from the Rpi/Esp and finally the output to the LED strip. Or you can use sacrifice LED level shifter from the tutorial on my blog if you don't have an access to SN74AHCT125N but it's a bit more complicated.

**Reply by BassXX** [reply]
Yes the example animations with the LED strip do always work. The USB Grabber is always passing through HDMI Signal, but the ambilight is working only random.

Thanks, will check and then try to build the level shifter and try again!

**Reply by awawa-dev** [reply]
So I suggest you to fix the power supply first. The level shifter may not be necessary in your case after all, but you can verify it later. Two HyperHDR logs would help clarify the situation with USB grabber: one taken when it works and the second one when it doesn't.

**Reply by BassXX** [reply]
Ok will do and Will report when done. Thank you

**Reply by BassXX** [reply]
Hey there! Just supplied the Pi with its own PSU.

However, same Problem persists. On bootup the leds are on and also I can use the idle effects without problem. But the ambilight is not working with my USB grabber. I have provided common ground on pin 20 and the leds are controlled via SPI pin 19. any idea what to try next? here are some bootup logs:


Hyper HDR


‭2022-08-17T17:40:45.418Z [MAIN] Database path: '/.hyperhdr/db/hyperhdr.db', readonlyMode = disabled‬
‭2022-08-17T17:40:45.418Z [MAIN] Starting HyperHdr - 17.0.0.0, (HEAD detached at dec81c0) (Awawa-2a2ed8d/dec81c0-1631541363), built: Sep 15 2021:16:53:51‬
‭2022-08-17T17:40:45.419Z [MAIN] Set user data path to '/.hyperhdr'‬
‭2022-08-17T17:40:45.481Z [DB] Database opened: /.hyperhdr/db/hyperhdr.db‬
‭2022-08-17T17:40:45.673Z [SETTINGSMGR] Settings database initialized‬
‭2022-08-17T17:40:45.675Z [HYPERHDR] Sound device is disabled‬
‭2022-08-17T17:40:45.779Z [V4L2:AUTO] Cropping image: width=0 height=0; crop: left=0 right=0 top=0 bottom=0 ‬
‭2022-08-17T17:40:45.780Z [V4L2:AUTO] Found capture device: /dev/video15‬
‭2022-08-17T17:40:45.781Z [V4L2:AUTO] Found capture device: /dev/video21‬
‭2022-08-17T17:40:45.783Z [V4L2:AUTO] Found capture device: /dev/video14‬
‭2022-08-17T17:40:45.784Z [V4L2:AUTO] Found capture device: /dev/video22‬
‭2022-08-17T17:40:45.850Z [V4L2:AUTO] Found capture device: /dev/video0‬
‭2022-08-17T17:40:45.852Z [V4L2:AUTO] USB3.0 HD Video Capture: USB3.0 (video0) [/dev/video0] 1920 x 1080 @ 60 fps mjpeg‬
‭2022-08-17T17:40:45.852Z [V4L2:AUTO] USB3.0 HD Video Capture: USB3.0 (video0) [/dev/video0] 1920 x 1080 @ 30 fps mjpeg‬
‭2022-08-17T17:40:45.852Z [V4L2:AUTO] USB3.0 HD Video Capture: USB3.0 (video0) [/dev/video0] 1280 x 720 @ 60 fps mjpeg‬
‭2022-08-17T17:40:45.852Z [V4L2:AUTO] USB3.0 HD Video Capture: USB3.0 (video0) [/dev/video0] 1280 x 720 @ 30 fps mjpeg‬
‭2022-08-17T17:40:45.853Z [V4L2:AUTO] USB3.0 HD Video Capture: USB3.0 (video0) [/dev/video0] 960 x 540 @ 60 fps mjpeg‬
‭2022-08-17T17:40:45.853Z [V4L2:AUTO] USB3.0 HD Video Capture: USB3.0 (video0) [/dev/video0] 960 x 540 @ 30 fps mjpeg‬
‭2022-08-17T17:40:45.854Z [V4L2:AUTO] USB3.0 HD Video Capture: USB3.0 (video0) [/dev/video0] 800 x 450 @ 60 fps mjpeg‬
‭2022-08-17T17:40:45.854Z [V4L2:AUTO] USB3.0 HD Video Capture: USB3.0 (video0) [/dev/video0] 800 x 450 @ 30 fps mjpeg‬
‭2022-08-17T17:40:45.854Z [V4L2:AUTO] USB3.0 HD Video Capture: USB3.0 (video0) [/dev/video0] 640 x 360 @ 60 fps mjpeg‬
‭2022-08-17T17:40:45.855Z [V4L2:AUTO] USB3.0 HD Video Capture: USB3.0 (video0) [/dev/video0] 640 x 360 @ 30 fps mjpeg‬
‭2022-08-17T17:40:45.856Z [V4L2:AUTO] Restarting video grabber is now blocked due to reloading of the configuration‬
‭2022-08-17T17:40:45.856Z [V4L2:auto] (GrabberWrapper.cpp:415) Auto resume is: enabled‬
‭2022-08-17T17:40:45.856Z [V4L2:AUTO] Cropping image: width=0 height=0; crop: left=0 right=0 top=0 bottom=0 ‬
‭2022-08-17T17:40:45.857Z [V4L2:auto] (GrabberWrapper.cpp:42) CEC keycode. Start: 0, stop: 0‬
‭2022-08-17T17:40:45.857Z [V4L2:AUTO] (Grabber.cpp:202) setBrightnessContrastSaturationHue nothing changed‬
‭2022-08-17T17:40:45.857Z [V4L2:AUTO] (V4L2Grabber.cpp:98) setHdrToneMappingMode to: Disabled‬
‭2022-08-17T17:40:45.858Z [V4L2:AUTO] (Grabber.cpp:140) setFpsSoftwareDecimation to: 1‬
‭2022-08-17T17:40:45.858Z [SIGNAL_OLD] (DetectionManual.cpp:85) Signal detection area set to: 0.250000,0.250000 x 0.750000,0.750000‬
‭2022-08-17T17:40:45.859Z [SIGNAL_OLD] (DetectionManual.cpp:75) Signal threshold set to: {12, 12, 12} and frames: 200‬
‭2022-08-17T17:40:45.859Z [V4L2:AUTO] Automatic signal detection is now enabled‬
‭2022-08-17T17:40:45.859Z [SIGNAL_AUTO] (DetectionAutomatic.cpp:368) Automatic signal detection -> errorTolerance: 9, modelTolerance: 90, sleepTime: 5000, wakeTime: 1000‬
‭2022-08-17T17:40:45.860Z [V4L2:AUTO] (Grabber.cpp:307) setDeviceVideoStandard preparing to restart video grabber. Old: '' new: 'auto'‬
‭2022-08-17T17:40:45.860Z [V4L2:AUTO] Delayed restart of the grabber due to change of selected device‬
‭2022-08-17T17:40:45.861Z [V4L2:AUTO] (Grabber.cpp:158) Force encoding to: NO_CHANGE (old: NO_CHANGE)‬
‭2022-08-17T17:40:45.861Z [V4L2:AUTO] setQFrameDecimation is now: disabled‬
‭2022-08-17T17:40:45.868Z [X11_SYSTEM:AUTO] Cropping image: width=0 height=0; crop: left=0 right=0 top=0 bottom=0 ‬
‭2022-08-17T17:40:45.877Z [X11_SYSTEM:AUTO] Loaded X11 proxy library for screen capturing‬
‭2022-08-17T17:40:45.883Z [X11_SYSTEM:AUTO] Restarting video grabber is now blocked due to reloading of the configuration‬
‭2022-08-17T17:40:45.883Z [X11_SYSTEM:AUTO] (Grabber.cpp:117) Set new width: 512, height: 512 for capture‬
‭2022-08-17T17:40:45.884Z [X11_SYSTEM:AUTO] (Grabber.cpp:262) setWidthHeight preparing to restarting video grabber 0‬
‭2022-08-17T17:40:45.884Z [X11_SYSTEM:AUTO] Delayed restart of the grabber due to change of the capturing resolution‬
‭2022-08-17T17:40:45.884Z [X11_SYSTEM:AUTO] Delayed restart of the grabber due to change of the framerate‬
‭2022-08-17T17:40:45.884Z [SIGNAL_OLD] (DetectionManual.cpp:85) Signal detection area set to: 0.250000,0.250000 x 0.750000,0.750000‬
‭2022-08-17T17:40:45.884Z [SIGNAL_OLD] (DetectionManual.cpp:75) Signal threshold set to: {0, 0, 0} and frames: 50‬
‭2022-08-17T17:40:45.884Z [X11_SYSTEM:AUTO] (Grabber.cpp:307) setDeviceVideoStandard preparing to restart video grabber. Old: '' new: 'auto'‬
‭2022-08-17T17:40:45.884Z [X11_SYSTEM:AUTO] Delayed restart of the grabber due to change of selected device‬
‭2022-08-17T17:40:45.987Z [SETTINGSMGR] Settings database initialized‬
‭2022-08-17T17:40:45.992Z [BLACKBORDER] Set mode to: default‬
‭2022-08-17T17:40:45.992Z [COMPONENTREG0] Blackborder detector: enabled‬
‭2022-08-17T17:40:45.993Z [IMAGETOLED0] (ImageProcessor.cpp:153) set user led mapping to advanced‬
‭2022-08-17T17:40:45.993Z [IMAGETOLED0] (ImageProcessor.cpp:136) setSparseProcessing to 0‬
‭2022-08-17T17:40:45.995Z [RGB_TRANSFORM0] RGB transform classic_config: 1, saturationGain: 1.000000, luminanceGain: 1.000000, backlightThreshold: 1‬
‭2022-08-17T17:40:45.995Z [RGB_TRANSFORM0] setBacklightThreshold: 1‬
‭2022-08-17T17:40:45.995Z [ADJUST_RED0] Set correction to 255‬
‭2022-08-17T17:40:45.996Z [ADJUST_BLUE0] Set correction to 255‬
‭2022-08-17T17:40:45.996Z [ADJUST_GREEN0] Set correction to 255‬
‭2022-08-17T17:40:46.044Z [JSONSERVER] (JsonServer.cpp:28) Created instance‬
‭2022-08-17T17:40:46.057Z [JSONSERVER] Started on port 19444‬
‭2022-08-17T17:40:46.000Z [DB] Database opened: /.hyperhdr/db/hyperhdr.db‬
‭2022-08-17T17:40:46.057Z [LEDDEVICE] Start LedDevice 'sk6812spi'.‬
‭2022-08-17T17:40:46.057Z [COMPONENTREG0] Smoothing: enabled‬
‭2022-08-17T17:40:46.057Z [SMOOTHING0] Creating config (0) => type: Alternative, dirMode: false, pause: false, settlingTime: 150ms, interval: 20ms (50Hz), antiFlickTres: 0, antiFlickStep: 0, antiFlickTime: 0‬
‭2022-08-17T17:40:46.058Z [SMOOTHING0] Clearing queued colors‬
‭2022-08-17T17:40:46.058Z [SMOOTHING0] Selecting config (0) => type: Alternative, dirMode: false, pause: false, settlingTime: 150ms, interval: 20ms (50Hz), antiFlickTres: 0, antiFlickStep: 0, antiFlickTime: 0‬
‭2022-08-17T17:40:46.058Z [SMOOTHING0] Selecting config (0) => type: Alternative, dirMode: false, pause: false, settlingTime: 150ms, interval: 20ms (50Hz), antiFlickTres: 0, antiFlickStep: 0, antiFlickTime: 0‬
‭2022-08-17T17:40:46.058Z [LEDDEVICE] (LedDevice.cpp:149) deviceConfig: [{"colorOrder":"grb","currentLedCount":211,"hardwareLedCount":211,"invert":false,"latchTime":0,"output":"/dev/spidev0.0","rate":3000000,"rewriteTime":1000,"type":"sk6812spi","whiteAlgorithm":"subtract_minimum"}]‬
‭2022-08-17T17:40:46.061Z [LEDDEVICE] (LedDevice.cpp:408) LatchTime updated to 0ms‬
‭2022-08-17T17:40:46.061Z [LEDDEVICE] (LedDevice.cpp:429) Refresh interval = 1000ms‬
‭2022-08-17T17:40:46.062Z [LEDDEVICE] (LedDevice.cpp:435) RewriteTime updated to 1000ms‬
‭2022-08-17T17:40:46.062Z [LEDDEVICE] (ProviderSpi.cpp:47) _baudRate_Hz [3000000], _latchTime_ms [0], _spiType: ‬
‭2022-08-17T17:40:46.063Z [LEDDEVICE] (ProviderSpi.cpp:48) _spiDataInvert [0], _spiMode [0]‬
‭2022-08-17T17:40:46.063Z [LEDDEVICE] (LedDeviceSk6812SPI.cpp:41) whiteAlgorithm : subtract_minimum‬
‭2022-08-17T17:40:46.065Z [FLATBUFCONN] Connecting to HyperHDR: 127.0.0.1:19401‬
‭2022-08-17T17:40:46.068Z [MUXER0] Register new input 'System/VIDEOGRABBER' with priority 253 as inactive‬
‭2022-08-17T17:40:46.068Z [COMPONENTREG0] Video capture device: enabled‬
‭2022-08-17T17:40:46.090Z [V4L2:AUTO] Multithreading for V4L2 is enabled. Available thread's count 4‬
‭2022-08-17T17:40:46.091Z [V4L2:AUTO] Found capture device: /dev/video15‬
‭2022-08-17T17:40:46.093Z [V4L2:AUTO] Found capture device: /dev/video21‬
‭2022-08-17T17:40:46.095Z [V4L2:AUTO] Found capture device: /dev/video14‬
‭2022-08-17T17:40:46.096Z [V4L2:AUTO] Found capture device: /dev/video22‬
‭2022-08-17T17:40:46.097Z [V4L2:AUTO] Found capture device: /dev/video0‬
‭2022-08-17T17:40:46.098Z [V4L2:AUTO] (V4L2Grabber.cpp:165) Forcing auto discovery device‬
‭2022-08-17T17:40:46.099Z [V4L2:AUTO] (V4L2Grabber.cpp:171) Auto discovery set to USB3.0 HD Video Capture: USB3.0 (video0)‬
‭2022-08-17T17:40:46.099Z [V4L2:AUTO] (V4L2Grabber.cpp:185) Searching for USB3.0 HD Video Capture: USB3.0 (video0) 0 x 0 @ 0 fps, input: -1 (NO_CHANGE)‬
‭2022-08-17T17:40:46.100Z [V4L2:AUTO] Forcing best guess‬
‭2022-08-17T17:40:46.100Z [V4L2:AUTO] *************************************************************************************************‬
‭2022-08-17T17:40:46.100Z [V4L2:AUTO] Starting V4L2 grabber. Selected: USB3.0 HD Video Capture: USB3.0 (video0) [/dev/video0] 640 x 360 @ 30 fps mjpeg‬
‭2022-08-17T17:40:46.100Z [V4L2:AUTO] *************************************************************************************************‬
‭2022-08-17T17:40:46.101Z [V4L2:AUTO] (V4L2Grabber.cpp:780) Hardware cropping is not supported: ignoring‬
‭2022-08-17T17:40:46.101Z [V4L2:AUTO] Set device input to: Camera 1‬
‭2022-08-17T17:40:46.103Z [V4L2:AUTO] Set resolution to: 640 x 360‬
‭2022-08-17T17:40:46.105Z [V4L2:AUTO] Set framerate to 30 FPS‬
‭2022-08-17T17:40:46.105Z [V4L2:AUTO] (V4L2Grabber.cpp:78) LUT folder location: '/usr/share/hyperhdr/lut'‬
‭2022-08-17T17:40:46.106Z [V4L2:AUTO] Video pixel format is set to: MJPEG‬
‭2022-08-17T17:40:46.111Z [V4L2:AUTO] Started‬
‭2022-08-17T17:40:46.070Z [BOBLIGHT] Instance created‬
‭2022-08-17T17:40:46.113Z [HYPERMANAGER] HyperHDR instance 'My Leds LGTV' has been started‬
‭2022-08-17T17:40:46.113Z [DAEMON] Request CEC‬
‭2022-08-17T17:40:46.113Z [DAEMON] Opening libCEC library.‬
‭2022-08-17T17:40:46.114Z [CEC] CEC object created‬
‭2022-08-17T17:40:46.114Z [CEC] Starting CEC handler.‬
‭2022-08-17T17:41:22.816Z [CEC] Success. CEC adapter type 'Linux' is ready (/dev/cec0, /sys/bus/cec/devices/cec0)‬
‭2022-08-17T17:41:22.820Z [DAEMON] Success: libCEC library loaded.‬
‭2022-08-17T17:40:46.071Z [COMPONENTREG0] LED device: enabled‬
‭2022-08-17T17:40:46.072Z [EFFECTENGINE0] Run effect "Rainbow swirl" on channel 0‬
‭2022-08-17T17:40:46.077Z [FLATBUFSERVER] Started on port 19400‬
‭2022-08-17T17:40:46.078Z [PROTOSERVER] Started on port 19445‬
‭2022-08-17T17:40:46.085Z [WEBSERVER] Initialize Webserver‬
‭2022-08-17T17:40:46.087Z [WEBSERVER] Initialize Webserver‬
‭2022-08-17T17:40:46.125Z [EFFECTENGINE0] (EffectEngine.cpp:152) Start the effect: name [Rainbow swirl], smoothCfg [32]‬
‭2022-08-17T17:40:46.126Z [MUXER0] Register new input 'System/EFFECT' with priority 0 as inactive‬
‭2022-08-17T17:40:46.126Z [HYPERHDR] Initial foreground effect 'Rainbow swirl' started‬
‭2022-08-17T17:40:46.127Z [SMOOTHING0] Using alternative smoothing input (0)‬
‭2022-08-17T17:40:46.127Z [SMOOTHING0] Enabling timer. Now timer is active: 1, remaining time to run: 20‬
‭2022-08-17T17:40:46.147Z [SMOOTHING0] Using alternative smoothing procedure (0)‬
‭2022-08-17T17:40:46.191Z [EFFECT0(Rainbo...)] Begin playing the effect with priority: 0‬
‭2022-08-17T17:40:46.216Z [MUXER0] Priority 0 is now active‬
‭2022-08-17T17:40:46.217Z [MUXER0] Set visible priority to 0‬
‭2022-08-17T17:40:46.217Z [IMAGETOLED0] Total index number is: 1000. Sparse processing: disabled, image size: 80 x 45, area number: 211‬
‭2022-08-17T17:40:46.218Z [IMAGETOLED0] Total index number is: 1000. Sparse processing: disabled, image size: 80 x 45, area number: 211‬
‭2022-08-17T17:40:46.218Z [SMOOTHING0] Clearing queued colors‬
‭2022-08-17T17:40:46.219Z [SMOOTHING0] Selecting config (32) => type: Linear, dirMode: false, pause: false, settlingTime: 200ms, interval: 40ms (25Hz), antiFlickTres: 0, antiFlickStep: 0, antiFlickTime: 0‬
‭2022-08-17T17:40:46.219Z [SMOOTHING0] Using linear smoothing input (32)‬
‭2022-08-17T17:40:46.219Z [SMOOTHING0] Enabling timer. Now timer is active: 1, remaining time to run: 40‬
‭2022-08-17T17:40:46.219Z [HYPERHDR0] New priority[0], previous [255]‬
‭2022-08-17T17:40:46.219Z [HYPERHDR0] New source available -> switch LED-Device on‬
‭2022-08-17T17:40:46.220Z [IMAGETOLED0] (ImageProcessor.cpp:180) set hard led mapping to multicolor_mean‬
‭2022-08-17T17:40:46.259Z [SMOOTHING0] Using linear smoothing procedure (32)‬
‭2022-08-17T17:40:46.277Z [CEC] CEC library warning: libCEC has not been compiled with support for the Pulse-Eight USB-CEC Adapter‬
‭2022-08-17T17:40:46.278Z [CEC] CEC library warning: CLinuxCECAdapterCommunication::Open - physical address is invalid‬
‭2022-08-17T17:40:46.354Z [WEBSERVER] Apply Webserver settings‬
‭2022-08-17T17:40:46.354Z [WEBSERVER] Apply Webserver settings‬
‭2022-08-17T17:40:46.355Z [WEBSERVER] Set document root to: :/webconfig‬
‭2022-08-17T17:40:46.355Z [WEBSERVER] Set document root to: :/webconfig‬
‭2022-08-17T17:40:46.357Z [WEBSERVER] Started on port 8090 name 'HyperHDR Webserver'‬
‭2022-08-17T17:40:46.388Z [WEBSERVER] Setup SSL certificate‬
‭2022-08-17T17:40:46.389Z [WEBSERVER] Setup private SSL key‬
‭2022-08-17T17:40:46.389Z [WEBSERVER] Started on port 8092 name 'HyperHDR Webserver'‬
‭2022-08-17T17:40:49.238Z [EFFECT0(Rainbo...)] The effect quits with priority: 0‬
‭2022-08-17T17:40:49.240Z [MUXER0] Removed source priority 0‬
‭2022-08-17T17:40:49.240Z [MUXER0] Switching from color effect. Waiting till: 19:40:52‬
‭2022-08-17T17:40:49.240Z [EFFECTENGINE0] Effect 'Rainbow swirl' has finished.‬
‭2022-08-17T17:40:49.240Z [EFFECT0(Rainbo...)] Deleting effect named: 'Rainbow swirl'‬
‭2022-08-17T17:40:52.312Z [MUXER0] Set visible priority to 255‬
‭2022-08-17T17:40:52.312Z [SMOOTHING0] Clearing queued colors‬
‭2022-08-17T17:40:52.312Z [SMOOTHING0] Selecting config (0) => type: Alternative, dirMode: false, pause: false, settlingTime: 150ms, interval: 20ms (50Hz), antiFlickTres: 0, antiFlickStep: 0, antiFlickTime: 0‬
‭2022-08-17T17:40:52.313Z [SMOOTHING0] Using alternative smoothing input (0)‬
‭2022-08-17T17:40:52.313Z [SMOOTHING0] Enabling timer. Now timer is active: 1, remaining time to run: 20‬
‭2022-08-17T17:40:52.313Z [HYPERHDR0] New priority[255], previous [0]‬
‭2022-08-17T17:40:52.313Z [HYPERHDR0] No source left -> switch LED-Device off‬
‭2022-08-17T17:40:52.314Z [IMAGETOLED0] (ImageProcessor.cpp:180) set hard led mapping to multicolor_mean‬
‭2022-08-17T17:40:52.314Z [IMAGETOLED0] Total index number is: 1000. Sparse processing: disabled, image size: 80 x 45, area number: 211‬
‭2022-08-17T17:40:52.334Z [SMOOTHING0] Using alternative smoothing procedure (0)‬
‭2022-08-17T17:40:55.323Z [CEC] CEC library signaling error: CLinuxCECAdapterCommunication::Write - ioctl CEC_TRANSMIT failed - tx_status=00 errno=64‬
‭2022-08-17T17:40:55.323Z [CEC] CEC library signaling error: CLinuxCECAdapterCommunication::Write - ioctl CEC_TRANSMIT failed - tx_status=00 errno=64‬
‭2022-08-17T17:41:10.833Z [WEBSOCKET] (WebSocketClient.cpp:30) New connection from ::ffff:192.168.0.94‬
‭2022-08-17T17:41:10.834Z [WEBSOCKET] (JsonAPI.cpp:89) Client '::ffff:192.168.0.94' switch to HyperHDR instance 0‬
‭2022-08-17T17:41:13.600Z [CEC] CEC library signaling error: CLinuxCECAdapterCommunication::Write - ioctl CEC_TRANSMIT failed - tx_status=00 errno=64‬
‭2022-08-17T17:41:13.601Z [CEC] CEC library signaling error: CLinuxCECAdapterCommunication::Write - ioctl CEC_TRANSMIT failed - tx_status=00 errno=64‬
‭2022-08-17T17:41:23.002Z [WEBSOCKET] (WebSocketClient.cpp:204) send close: 1000 ‬
‭2022-08-17T17:41:23.364Z [WEBSOCKET] (WebSocketClient.cpp:30) New connection from ::ffff:192.168.0.94‬
‭2022-08-17T17:41:23.365Z [WEBSOCKET] (JsonAPI.cpp:89) Client '::ffff:192.168.0.94' switch to HyperHDR instance 0‬
‭2022-08-17T17:41:23.511Z [DB] Database opened: /.hyperhdr/db/hyperhdr.db‬
‭2022-08-17T17:41:23.983Z [WEBSOCKET] (JsonAPI.cpp:1300) log streaming activated for client ::ffff:192.168.0.94‬

**Reply by awawa-dev** [reply]
1 Does the grabber is receiving a valid (and supported by the grabber) video signal at the boot? For test, set 1080p60 resolution that every grabber should handle. The signal should be present before you turn the Rpi on. Some grabbers are picky about that.
2 when that happens after the boot, could you ssh to your Raspberry and execute ```dmesg``` to get system logs?

**Reply by awawa-dev** [reply]
3 also make sure that the video signal isn't protected by HDCP

**Reply by BassXX** [reply]
Thats very interesting. I set the resolution on my Win PC to 1080p and now it works without problems. Also on all the other inputs. Afterwards i went back to 4k and it keeps working. 
Also when i boot, the HDMI must be my Win PC. If i boot and my (other) Raspberry Pi with Kodi is active on HDMI, it wont work.

Must be the USB grabber then probably?

I will observe the behavior over the next days. If i notice anything weird, i will contact you for help. 
Thanks for now, you helped me a lot :)

**Reply by awawa-dev** [reply]
Some refresh rates/colorspaces may not be supported by the grabber. 23/24Hz are especially problematic. And there are few reports that Kodi may select the best output for the movie that won't be recognized by some grabbers. Workaround is to set the resolution/refresh rate manually and disable the auto-selection.

**Reply by BassXX** [reply]
Can you recommend a good USB Grabber/Capture card which has proven to work well?

And whats the deal with the level shifter and SK6812? Do I still need it?
Regards

**Reply by awawa-dev** [reply]
If it's always working without blinking then you don't need a level shifter. It's a bit outside of the technical specification for sk6812 so may not be stable all the time. Some other factories can play a role if it will work or not without the level shifter like: cables length that affects voltages and finally data signal and +5v real voltage level.

I don't use Rpi for playing movies and use a PC instead. So I didn't test the grabber for such scenario. Otherwise even cheapest black Rullz MS2109 USB2.0 works fine and doesn't need a video source to be present at the boot. 23/24Hz also works for 1080p but I think there are reports about problems with Rpi/Kodi for 4k@23/24 (for example https://github.com/awawa-dev/HyperHDR/discussions/315 the output is characteristic for MS2109 so probable that's the case).
